ids/short_node_id.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package ids
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/staking"
+	"github.com/ava-labs/avalanchego/utils"
+	"github.com/ava-labs/avalanchego/utils/hashing"
+)
+
+const (
+	ShortNodeIDPrefix = "NodeID-"
+	ShortNodeIDLen    = ShortIDLen
+)
+
+var (
+	EmptyShortNodeID = ShortNodeID{}
+
+	errShortNodeID = errors.New("insufficient ShortNodeID length")
+
+	_ utils.Sortable[ShortNodeID] = ShortNodeID{}
+)
+
+type ShortNodeID ShortID
```

Comment:
If we have the generic string-based node id why do we need this one? My understanding is that this is the "legacy" node id we use currently, I think what we want instead is a function that turns a cert into the string-based node id.

Reply:
The serialization of ShortNodeID is different from the serialization of NodeID.
So I have to keep ShortNodeID around to keep backward compatibility with genesis and some P-chain transactions (to add validator/delegators). Apart from these two cases, we'll use NodeID everywhere else